ex03.exs

```diff
@@ -60,9 +60,18 @@ defmodule Ex03 do
   """
 
   def pmap(collection, process_count, function) do
-    « your code here »
+    collection = collection |> Enum.into([])
+    chunk_size = ( Enum.count( collection) / process_count ) |> round
+    chunks = Enum.chunk_every( collection, chunk_size)
+    fmap = &( Task.async( fn -> Enum.map( &1, function) end ) )
+    tasks = Enum.map( chunks, fmap )
```

Comment:
## -5 for style/complexity

I'm uncomfortable with the number of variables, and the general linear nature of this code. 

I'd rather have seen something like:

~~~ elixir
  defp into_chunks(collection, count) do
    chunk_size = div(Enum.count( collection), count )
    Enum.chunk_every( collection, chunk_size)
  end

  def pmap(collection, process_count, function) do
    chunk_mapper = &( Task.async( fn -> Enum.map( &1, function) end ) )

    collection
    |> into_chunks(process_count)
    |> Enum.map( chunk_mapper )
    |> Enum.map( &( Task.await( &1 ) ) )
    |> Enum.concat()
  end
~~~